hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,55 @@
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def migration
+  export_dir = '/var/tmp/engine_upgrade'
+  mongo_dir = '/var/lib/mongodb'
+  hiera_file = '/etc/foreman-installer/custom-hiera.yaml'
+  mongo_backup = '/var/tmp/mongo_backup'
+  mongo_conf = '/etc/opt/rh/rh-mongodb34/mongod.conf'
+
+  # Verify we have a good value in the answerfile for the MongoDB DB name.
+  pulp_db_param = param('katello', 'pulp_db_name') || param('foreman_proxy_content', 'pulp_db_name')
+  fail_and_exit('Unknown pulp database name') unless pulp_db_param
```

Comment:
This check probably belongs in `pre_validation`.

Reply:
Done